server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/RandomStoring.java

```diff
@@ -0,0 +1,121 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.mail.MessagingException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.User;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.metrics.api.MetricFactory;
+import org.apache.james.transport.mailets.delivery.MailDispatcher;
+import org.apache.james.transport.mailets.delivery.MailboxAppender;
+import org.apache.james.transport.mailets.delivery.SimpleMailStore;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMailet;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.Streams;
+
+/**
+ * Receives a Mail from the Queue and takes care of delivery of the
+ * message to local inboxes.
+ *
+ * This mailet is a composition of RecipientRewriteTable, SieveMailet
+ * and MailboxManager configured to mimic the old "LocalDelivery"
+ * James 2.3 behavior.
+ */
+public class RandomStoring extends GenericMailet {
+
+    private static final String RANDOM_STORING_MAILS_METRIC_NAME = "randomStoringMails";
+    private static final int MIN_NUMBER_OF_RECIPIENTS = 4;
+    private static final int MAX_NUMBER_OF_RECIPIENTS = 8;
+    private final UsersRepository usersRepository;
+    private final MailboxManager mailboxManager;
+    private final MetricFactory metricFactory;
+    private final Iterator<Integer> randomSizes;
+    private MailDispatcher mailDispatcher;
+
+    @Inject
+    public RandomStoring(UsersRepository usersRepository, @Named("mailboxmanager") MailboxManager mailboxManager,
+                         MetricFactory metricFactory) {
+        this.metricFactory = metricFactory;
+        this.usersRepository = usersRepository;
+        this.mailboxManager = mailboxManager;
+        this.randomSizes = new Random().ints(MIN_NUMBER_OF_RECIPIENTS, MAX_NUMBER_OF_RECIPIENTS + 1).boxed().iterator();
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        try {
+            mail.setRecipients(generateRandomUsers());
+            mailDispatcher.dispatch(mail);
```

Comment:
you are also supposed to choose a folder randomly (each user should have 100 folders).
For that, you can set this attribute: `AttributeName.of(DELIVERY_PATH_PREFIX + username)` to the folder name and you're done.